src/Uno.UI/Microsoft/UI/Xaml/Controls/ScrollPresenter/SnapPoint.cs

```diff
@@ -0,0 +1,1920 @@
+
+	private SnapPointApplicableRangeType ApplicableRangeType()
+	{
+		return m_applicableRangeType;
+	}
+#endif
+
+#if DEBUG
+	internal Color VisualizationColor { get; set; } = Colors.Black;
+#endif // DBG
+
+	// Returns True if this snap point snaps around the provided value.
+	internal bool SnapsAt(
+		(double, double) actualApplicableZone,
+		double value)
+	{
+		if (actualApplicableZone.Item1 <= value &&
+			actualApplicableZone.Item2 >= value)
+		{
+			double snappedValue = Evaluate(actualApplicableZone, (float)value);
+
+			return Math.Abs(value - snappedValue) < s_equalityEpsilon;
+		}
+
+		return false;
+	}
+
+	// Uno docs: This doesn't seem to be used.
+	//void SetBooleanParameter(
+	//	ExpressionAnimation expressionAnimation,
+	//	string booleanName,
+	//	bool booleanValue)
+	//{
+	//	// SCROLLPRESENTER_TRACE_VERBOSE(nullptr, TRACE_MSG_METH_STR_INT, METH_NAME, this, booleanName, booleanValue);
+
+	//	expressionAnimation.SetBooleanParameter(booleanName, booleanValue);
+	//}
+
+	internal void SetScalarParameter(
+		ExpressionAnimation expressionAnimation,
+		string scalarName,
+		float scalarValue)
+	{
+		// SCROLLPRESENTER_TRACE_VERBOSE(nullptr, TRACE_MSG_METH_STR_FLT, METH_NAME, this, scalarName, scalarValue);
+
+		expressionAnimation.SetScalarParameter(scalarName, scalarValue);
+	}
+}
+
+public partial class ScrollSnapPointBase : SnapPointBase
+{
+	private protected double m_alignmentAdjustment = 0.0; // Non-zero adjustment based on viewport size, when the alignment is Center or Far.
+
+	/////////////////////////////////////////////////////////////////////
+	/////////////////      Scroll Snap Points     ///////////////////////
+	/////////////////////////////////////////////////////////////////////
+
+	// Required for Modern Idl bug, should never be called.
+	internal ScrollSnapPointBase()
+	{
+		// throw (ERROR_CALL_NOT_IMPLEMENTED);
+	}
+
+	public ScrollSnapPointsAlignment Alignment { get; private protected set; } = ScrollSnapPointsAlignment.Near;
+
+
+	// Returns True when this snap point is sensitive to the viewport size and is interested in future updates.
+	internal override bool OnUpdateViewport(double newViewport)
+	{
+		switch (Alignment)
+		{
+			case ScrollSnapPointsAlignment.Near:
+				MUX_ASSERT(m_alignmentAdjustment == 0.0);
+				return false;
+			case ScrollSnapPointsAlignment.Center:
+				m_alignmentAdjustment = -newViewport / 2.0;
+				break;
+			case ScrollSnapPointsAlignment.Far:
+				m_alignmentAdjustment = -newViewport;
+				break;
+		}
+		return true;
+	}
+}
+
+public partial class ScrollSnapPoint : ScrollSnapPointBase
+{
+	// Uno docs: In SnapPoint.h in WinUI code, the second parameter has a default value of ScrollSnapPointsAlignment::Near
+	// HOWEVER, the API seen in a real WinUI app says that the parameter is mandatory.
+	public ScrollSnapPoint(
+		double snapPointValue,
+		ScrollSnapPointsAlignment alignment)
+	{
+		Value = snapPointValue;
+		Alignment = alignment;
+	}
+
+#if ApplicableRangeType // UNO TODO:
```

Comment:
Is this from WInUI? Compiler variables are usually UPPER_SNAKE_CASED

Reply:
@dr1rrb Yes, from WinUI, and it was quite unclear to me if it should be true or false in our context. So left it as a TODO